llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -1280,6 +1280,83 @@ static MachineBasicBlock *emitSplitF64Pseudo(MachineInstr &MI,
   return BB;
 }
 
+static MachineBasicBlock *emitHardwareLoopMemcpy(MachineInstr &MI,
```

Comment:
I would restructure this so variables are a bit more local so its easier to read. For example create vregs just before you need them, and add comments to indicate the instruction you are intending to create.